proto/access.proto

```diff
@@ -79,6 +79,12 @@ message AccessAttempt {
   // Actor describes the entity attempting to gain access.
   // Optional.
   Actor actor = 3;
+
+  // The time the last successful access attempt was made.
+  google.protobuf.Timestamp last_successful_access_time = 4;
+
+  // The name of the zone that was last successfully accessed.
+  string last_successful_access_zone = 5;
```

Comment:
I'm still not sure what this field is for, when I should use it (as a driver or as a client), or why it's different from looking up the zone in the metadata of the access device. What is a `zone` in the context of an access device?

Reply:
It is something that is exposed through Gallagher's API, the zones are something defined on their system and is the last zone a cardholder successfully accessed

https://gallaghersecurity.github.io/cc-rest-docs/ref/cardholders.html

To be honest I was just trying to get as much useful stuff integrated as possible so we can show that we are surfacing data from their system.

Reply:
Yeah, that's fine, if it's a concept access devices have then we should support it. I'm just trying to make sure someone looking at this trait in 6 months doesn't get confused because SC also has the concept of zones.

Reading the Gallagher docs it looks like these properties belong on the Actor type rather than the AccessAttempt type. They seem to describe what the cardholder last did, which maps to Actor in the Access trait.

I'd be happy to add this to that type, could shorten the names slightly too:

```protobuf
message Actor {
 ...
  // The time this actor was last granted access.
  // The grant may have been performed by a different device to this one.
  google.protobuf.Timestamp last_grant_time = ?;

  // The access zone the actor was last granted access to.
  // The zone will be provided by the device and may not map to any Smart Core zone or device.
  string last_grant_zone = ?;
}
```

Something like that, does it make sense?

Reply:
You could see in the future also adding `last_seen_time` and other things like that to the actor message.

Reply:
Ah I missed the Actor field, thats where it belongs. I have moved the fields there and stuck them as 2 & 3 after the name as the stuff further down looked more like metadata